server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -548,11 +549,13 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
       batchTimeOut = 0;
     }
 
+    boolean yield = false;
+
     for (Range range : ranges) {
 
       boolean timesUp = batchTimeOut > 0 && System.nanoTime() > returnTime;
 
-      if (exceededMemoryUsage || tabletClosed || timesUp) {
+      if (exceededMemoryUsage || tabletClosed || timesUp || yield) {
         lookupResult.unfinishedRanges.add(range);
         continue;
```

Comment:
The way the Exception's javadoc reads, it would preempty the entire Scan RPC, but it's actually just pre-empting one Range. Is that intentional?